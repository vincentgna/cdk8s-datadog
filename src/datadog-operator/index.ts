// generated by import-crds.ts
// version 1.2.1
export * from "./datadoghq.com";