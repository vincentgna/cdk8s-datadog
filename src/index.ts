export class Hello {
  public sayHello() {
    return "hello, GoodNotes!";
  }
}
