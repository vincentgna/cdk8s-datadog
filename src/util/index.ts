export * from "./Types.generated";
export * from "./DatadogMonitorProps.generated";