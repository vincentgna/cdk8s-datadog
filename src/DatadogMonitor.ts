import { Construct } from "constructs";
import * as datadog from "./datadog-operator";
import { DatadogMonitorProps } from "./util";

/**
 * DatadogMonitor with strongly typed props
 */
export class DatadogMonitor extends Construct {
  constructor(scope: Construct, id: string, props: DatadogMonitorProps) {
    super(scope, id);

    // leave props.metadata undefined to have it generated by Chart:
    // https://github.com/cdk8s-team/cdk8s-core/blob/v2.67.2/src/chart.ts#L106-L131
    new datadog.DatadogMonitor(this, id, {
      metadata: props.kubeMetadata,
      spec: props,
    });
  }
}
