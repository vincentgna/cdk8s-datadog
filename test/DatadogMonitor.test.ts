import { Testing, Chart } from "cdk8s";
import { DatadogMonitor } from "../src";

test("DatadogMonitor props are strongly typed", () => {
  // GIVEN
  const app = Testing.app();
  const chart = new Chart(app, "test");
  new DatadogMonitor(chart, "Foo", {
    type: "metric alert",
    query: `avg(last_5m):sum:docker.cpu.usage{kube_deployment:chart-foo-deploy} > 80`,
    name: "[CDK8s] podinfo pod cpu utilisation is too high",
    message: `CPU Usage over 80 for the last 5min, review Autoscaling triggers`,
  });

  new DatadogMonitor(chart, "Bar", {
    kubeMetadata: {
      name: "bar",
    },
    type: "metric alert",
    query: `avg(last_5m):sum:docker.cpu.usage{kube_deployment:chart-foo-deploy} > 80`,
    name: "[CDK8s] podinfo pod cpu utilisation is too high",
    message: `CPU Usage over 80 for the last 5min, review Autoscaling triggers`,
  });
  // THEN
  expect(Testing.synth(chart)).toMatchSnapshot();
});
