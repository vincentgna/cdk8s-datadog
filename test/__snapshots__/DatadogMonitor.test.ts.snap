// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatadogMonitor props are strongly typed 1`] = `
[
  {
    "apiVersion": "datadoghq.com/v1alpha1",
    "kind": "DatadogMonitor",
    "metadata": {
      "name": "test-foo-c8c19855",
    },
    "spec": {
      "message": "CPU Usage over 80 for the last 5min, review Autoscaling triggers",
      "name": "[CDK8s] podinfo pod cpu utilisation is too high",
      "query": "avg(last_5m):sum:docker.cpu.usage{kube_deployment:chart-foo-deploy} > 80",
      "type": "metric alert",
    },
  },
  {
    "apiVersion": "datadoghq.com/v1alpha1",
    "kind": "DatadogMonitor",
    "metadata": {
      "name": "bar",
    },
    "spec": {
      "message": "CPU Usage over 80 for the last 5min, review Autoscaling triggers",
      "name": "[CDK8s] podinfo pod cpu utilisation is too high",
      "query": "avg(last_5m):sum:docker.cpu.usage{kube_deployment:chart-foo-deploy} > 80",
      "type": "metric alert",
    },
  },
]
`;
